### PR TITLE
[Workspace]Add "All use case" option to workspace form

### PR DESCRIPTION
**Pull request by wanglam**
### Description

The use case of workspace can only be one specific use case or all use case. This PR mainly includes below changes:
1. Modify multi use case selection to single and add "all use case" option
2. Avoid application been set to Inaccessible when workspace using all use case and application in any use case
3. Avoid nav group been hidden when workspace using all use case

### Issues Resolved

#7319 

## Screenshot

image

## Testing the changes

- Check out branch and Run `yarn osd boostrap`
- Add below config to the `opensearch_dashboards.yml`
```
workspace.enabled: true
uiSettings.overrides["home:useNewHomePage"]: true
```
- Run `yarn start --no-base-path` and login with admin user
- Go to [workspace create page](http://localhost:5601/app/workspace_create)
- Fill workspace unique name and select All use case
- Click create workspace and it will jump to workspace detail page like below image
image

(The "all use case" will be selected and the left navigation will changed to all style)

## Changelog

- feat: [Workspace]Add "All use case" option to workspace form

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/utils.ts [hunk 1]
```diff
@@ -32,14 +37,14 @@ export const getUseCaseFromFeatureConfig = (featureConfig: string) => {
...
-  const useCase = useCases.find(({ id }) => id === getUseCaseFromFeatureConfig(featureConfig));
-  if (useCase) {
-    return useCase.features.includes(featureId);
-  }
-  return false;
+  const availableFeatures =
+    useCaseId === ALL_USE_CASE_ID
+      ? useCases.reduce<string[]>((previous, { features }) => previous.concat(features ?? []), [])
+      : useCases.find(({ id }) => id === useCaseId)?.features ?? [];
+  return availableFeatures.includes(featureId);
```
Comment (Hailong-am) [inline src/plugins/workspace/public/utils.ts:47]: if an app have not register to any use case, will it be visible/accesable in all use case?

File: src/plugins/workspace/public/plugin.ts [hunk 2]
```diff
@@ -122,6 +124,7 @@ export class WorkspacePlugin implements Plugin<{}, {}, WorkspacePluginSetupDeps>
           if (
             navGroup.type !== NavGroupType.SYSTEM &&
             currentWorkspace.features &&
+            getFirstUseCaseOfFeatureConfigs(currentWorkspace.features) !== ALL_USE_CASE_ID &&
```
Comment (ruanyl) [inline src/plugins/workspace/public/plugin.ts:127]: Would be nice to have a comment on this logic

**Review by ruanyl — DISMISSED** [review]

File: src/plugins/workspace/public/utils.ts [hunk 3]
```diff
@@ -32,14 +37,14 @@ export const getUseCaseFromFeatureConfig = (featureConfig: string) => {
...
-  featureConfig: string,
+  useCaseId: string,
   useCases: WorkspaceUseCase[]
 ) => {
-  const useCase = useCases.find(({ id }) => id === getUseCaseFromFeatureConfig(featureConfig));
-  if (useCase) {
-    return useCase.features.includes(featureId);
-  }
-  return false;
+  const availableFeatures =
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/utils.ts:43]: I thought we will just bypass all the features in all use case. Why do we change that?

File: src/plugins/workspace/public/utils.ts (hunk 1 above)
Comment (wanglam) [reply src/plugins/workspace/public/utils.ts:47]: Based current implementation, the all use case will match all features inside registered use cases. If an app not register to any use case, it still can be accessible if it match below conditions:
1. `workspaceAvailability` is `outsideWorkspace` 
2. `navLinkStatus` equal `hidden`
3. `chromeless` equal `true`

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/utils.ts (hunk 3 above)
Comment (wanglam) [reply src/plugins/workspace/public/utils.ts:43]: Do you mean we don't need to check whether the application id exists in registered use cases? We add skip the feature verification logic outside for all use case workspace.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**